### ISSUE TITLE
Chat: Refactoring for style and modularity, First Half 

### ISSUE THREAD

**Issue by ebporter** [rifflearning/riff-rtc#72]
This story is about refactoring and modularizing all the video chat code.

Goals for (4/29 - 5/10) sprint:
- Refactor half the code for all of video chat, but still with some (introduced) bugs. But basically, be halfway through cleaned up code that can be worked on by anybody. 

Goals for (5/13 - 5/24) sprint:
- Refactor the second half the code for all of video chat, but still with some (introduced) bugs. But basically, be all the way through cleaned up code that can be worked on by anybody. 

Goals for (5/27 - 6/7) sprint:
- Clean up code for bugs and style and have a working version that is fully refactored. Be ready to start Phase 2, the packaging.

**Comment by juliariffgit** [comment]
Plugging away. Confident will hit the goal by the end of the sprint.

**Comment by juliariffgit** [comment]
Nicer code. Not fully finished, but at a good spot right now relative to before. This moved more quickly than anticipated. If Mike or Brec want to look at the branch, they could. But we could wait on this. Mike and Jordan to talk about the complicated plumbing with regard to redux store connects. They are aligned on strategy for connecting.

**Comment by juliariffgit** [comment]
Testing changes and fixing bugs (like screensharing). 

Is it good enough to put in develop/staging so we can continue testing there? Yes. 

Mike made some changes to meeting mediator yesterday, and we should ensure none of these changes affect the work that's been done already.

Jordan to create a pull request.

**Comment by juliariffgit** [comment]
PR now out for this. A lot of changes (approx 30 changed). Will take a while to review, but this is a first pass at breaking everything into components. Jordan will Also will install Zenhub add-on for and then link it to the issue after standup today. 

Jordan would like an in-person or chat 1:1 meeting for the review because there are so many.

**Comment by jordanreedie** [comment]
Mike and I met on monday, and we went over most of the changes. Ari has made some changes to the meeting mediator that I need to merge in with my changes, but after I do that this should get merged into develop.

**Comment by jordanreedie** [comment]
I merged Ari's changes & am just waiting on Mike to review and merge into develop

**Comment by ebporter** [comment]
This issue is closed.
